### PR TITLE
Improve dev env and enable local S3 testing

### PR DESCRIPTION
**Pull request by neolynx**
## Description of the Change

* Update and document the docker development environment
* Fix aws S3 depreciation warning in golangci-lint
  ```
   SA1019: config.WithEndpointResolverWithOptions is deprecated: The global endpoint resolution interface is deprecated. See deprecation docs on [WithEndpointResolver]. (staticcheck) 
   SA1019: aws.Endpoint is deprecated: This structure was used with the global [EndpointResolver] interface, which has been deprecated in favor of service-specific endpoint resolution. See the deprecation docs on that interface for more information.  (staticcheck)
   ```
* Allow enabling local S3 system tests by providing `aws.creds` file:
   ```
   AWS_ACCESS_KEY_ID=...
   AWS_SECRET_ACCESS_KEY=...
   ```
* Promote GitHub discussions in CONTRIBUTION,md
* use go 1.19 compatible golangci-lint

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ These are just guidelines, not rules. Use your best judgment, and feel free to p
 
 This project adheres to the Contributor Covenant [code of conduct](CODE_OF_CONDUCT.md).
 By participating, you are expected to uphold this code.
-Please report unacceptable behavior to [team@aptly.info](mailto:team@aptly.info).
+Please report unacceptable behavior to [aptlypublic@gmail.com](mailto:aptlypublic@gmail.com).
```
Comment (an0rak-dev) [inline CONTRIBUTING.md:14]: We don't have access to this mailing ? 

Should we not instead encourage the GitHub Discussion features instead ?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -11,7 +11,7 @@ These are just guidelines, not rules. Use your best judgment, and feel free to p
 
 This project adheres to the Contributor Covenant [code of conduct](CODE_OF_CONDUCT.md).
```
Comment (an0rak-dev) [inline CONTRIBUTING.md:12]: Unrelated but : on line 5 above, the aptly link seems invalid.

File: system/Dockerfile [hunk 3]
```diff
@@ -1,6 +1,6 @@
 FROM debian:bookworm-slim
 
-RUN apt-get update -y && apt-get install -y --no-install-recommends curl gnupg apg bzip2 xz-utils ca-certificates golang golang-go golang-doc golang-src make git python3 python3-requests-unixsocket python3-termcolor python3-swiftclient python3-boto python3-azure-storage g++ python3-etcd3 python3-plyvel && \
+RUN apt-get update -y && apt-get install -y --no-install-recommends curl gnupg apg bzip2 xz-utils ca-certificates golang golang-go golang-doc golang-src make git python3 python3-requests-unixsocket python3-termcolor python3-swiftclient python3-boto python3-azure-storage g++ python3-etcd3 python3-plyvel graphviz && \
```
Comment (an0rak-dev) [inline system/Dockerfile:3]: Why graphviz is needed now ?

**Review by an0rak-dev — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 2 above)
Comment (neolynx) [reply CONTRIBUTING.md:12]: fixed

**Review by an0rak-dev — APPROVED** [review]

**Pull request merged**